src/ui/atoms/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useCallback, useRef } from 'react'
+import './fileInput.scss'
+import classNames from 'classnames'
+import fileInputIcon from '../../../assets/images/file.png'
+import { Typography } from '../typography/Typography'
+import type { DeepReadonly } from 'superTypes'
+
+export type FileUploadProps = {
+  /**
+   * The main title
+   */
+  readonly label?: string
+  /**
+   * Context information for the user
+   */
+  readonly description?: JSX.Element
+  /**
+   * Drop one or several files
+   */
+  readonly multiple?: boolean
+  /**
+   * The list of the accepted formats
+   */
+  readonly acceptedFormats?: string[]
+  /**
+   * The size of the component
+   */
+  readonly size?: 'large' | 'medium' | 'small'
+  /**
+   * Make the component in an error state or not
+   */
+  readonly error?: boolean
+  /**
+   * An error message displayed when the component is in the error state
+   */
+  readonly errorMessage?: string
+  /**
+   * Callback method called when files are dropped
+   */
+  // lint rule bypassed in method props arguments
+  // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+  readonly onDropped: (files: File[]) => void
+}
+
+// lint rule bypassed while disabling the rule
+// eslint-disable-next-line max-lines-per-function
+export const FileInput: React.FC<FileUploadProps> = ({
+  label,
+  description,
+  multiple = true,
+  acceptedFormats,
+  size = 'medium',
+  error = false,
+  errorMessage,
+  onDropped
+}: DeepReadonly<FileUploadProps>): JSX.Element => {
+  const inputFileElement = useRef<HTMLDivElement>(null)
+  const draggingClassname = 'dragging'
+
+  const addDraggingClassName = useCallback((): void => {
+    inputFileElement.current?.classList.add(draggingClassname)
+  }, [])
+
+  const removeDraggingClassName = useCallback((): void => {
+    inputFileElement.current?.classList.remove(draggingClassname)
+  }, [])
+
+  type LabelDragEvent = React.DragEvent<HTMLLabelElement>
+
+  // lint rule bypassed because of type 'Element' is not compatible with readonly
+  // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+  const preventDefaultAndStopPropagation = useCallback((event: LabelDragEvent): void => {
+    event.preventDefault()
+    event.stopPropagation()
+  }, [])
+
+  const getFilesFromEvent = useCallback(
+    // lint rule bypassed because of type 'Element' is not compatible with readonly
+    // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+    (event: LabelDragEvent): File[] =>
+      Array.from(event.dataTransfer.items)
+        .map((item: DeepReadonly<DataTransferItem>) => item.getAsFile())
+        .filter((file: DeepReadonly<File> | null): file is File => !!file),
```

Comment:
I think we could simplify the filter function:
```suggestion
        .filter(Boolean),
```